pkg/netrc/netrc.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netrc
```

Comment:
It seems to me that a generic `credential` package would be handy with a netrc as an implementation.

Reply:
good, call i was trying to reduce the amount of changes introduced as i'm still new to the codebase. but i'll take a crack at that.

Reply:
Might want to drop in on the dev meeting Thursday and see how the maintainers feel about this sort of change if you haven't already.